x/wasm/internal/keeper/genesis_test.go

```diff
@@ -377,6 +391,116 @@ func TestFailFastImport(t *testing.T) {
 	}
 }
 
+func TestExportShouldNotContainContractCodeHistory(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "wasm")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+	ctx, keepers := CreateTestInput(t, false, tempDir, SupportedFeatures, nil, nil)
+	accKeeper, keeper := keepers.AccountKeeper, keepers.WasmKeeper
+
+	wasmCode, err := ioutil.ReadFile("./testdata/contract.wasm")
+	require.NoError(t, err)
+	var (
+		deposit = sdk.NewCoins(sdk.NewInt64Coin("denom", 100000))
+		creator = createFakeFundedAccount(ctx, accKeeper, deposit)
+		anyAddr = make([]byte, sdk.AddrLen)
+	)
+
+	firstCodeID, err := keeper.Create(ctx, creator, wasmCode, "https://github.com/CosmWasm/wasmd/blob/master/x/wasm/testdata/escrow.wasm", "", &types.AllowEverybody)
+	require.NoError(t, err)
+	secondCodeID, err := keeper.Create(ctx, creator, wasmCode, "https://github.com/CosmWasm/wasmd/blob/master/x/wasm/testdata/escrow.wasm", "", &types.AllowEverybody)
+	require.NoError(t, err)
+	initMsg := InitMsg{
+		Verifier:    anyAddr,
+		Beneficiary: anyAddr,
+	}
+	initMsgBz, err := json.Marshal(initMsg)
+	require.NoError(t, err)
+
+	// create instance
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 1)
+	contractAddr, err := keeper.Instantiate(ctx, firstCodeID, creator, creator, initMsgBz, "demo contract 1", nil)
+	require.NoError(t, err)
+
+	// and migrate to second code id
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() + 1)
+	_, err = keeper.Migrate(ctx, contractAddr, creator, secondCodeID, initMsgBz)
+	require.NoError(t, err)
+	// and contract contains 2 history elements
+	contractInfo := keeper.GetContractInfo(ctx, contractAddr)
+	require.NotNil(t, contractInfo)
+	require.Len(t, contractInfo.ContractCodeHistory, 2)
+	// when exported
+	state := ExportGenesis(ctx, keeper)
+	require.NoError(t, state.ValidateBasic())
+	require.Len(t, state.Contracts, 1)
+	assert.Len(t, state.Contracts[0].ContractInfo.ContractCodeHistory, 0)
+	assert.Nil(t, state.Contracts[0].ContractInfo.Created)
+}
+
+func TestImportContractWithCodeHistoryReset(t *testing.T) {
+	genesis := `
+{
+	"params":{
+		"code_upload_access": {
+			"permission": "Everybody"
+		},
+		"instantiate_default_permission": "Everybody"
+	},
+  "codes": [
+    {
+      "code_id": 1,
+      "code_info": {
+        "code_hash": "mrFpzGE5s+Qfn9Xe7OikXc0q169m7sMm4ZuV6pVA8Mc=",
+        "creator": "cosmos1qtu5n0cnhfkjj6l2rq97hmky9fd89gwca9yarx",
+        "source": "https://example.com",
+        "builder": "foo/bar:tag",
+        "instantiate_config": {
+          "type": 1,
```

Comment:
I thought you updated this to "Nobody" / "Everybody" instead of numbers? Or is that PR not merged?

Reply:
yes, this is an old configuration I had before. Good 👁️ . It also shows that the field is not validated proper I will fix this